Plugin catalog

jongough opened the pull request:
This pull implements local catalogs for the "built-in" plugins chartdldr_pi, wmm_pi and demo_pi_sample. This is a way to ensure translations in one plugin do not affect the translations in another plugin. This change is currently in ocpn_draw_pi and watchdog_pi.

This pull also changes the demo_pi_sample so that it can be built if OCPN_BUILD_SAMPLE is set to ON (-DOCPN_BUILD_SAMPLE=ON) and corrects issues with the build. The CMAKE files have been updated to reflect those used in chartdldr_pi and wmm_pi, so there should be consistency for a new developer.

bdbcat commented:
Jon...
I like the idea of a functional Demo Plugin that a new developer can use for reference.
But the rest  looks pretty scary.
Maybe split into more manageable bits?

Thanks
Dave

jongough commented:
Hi Dave,
   It took me a few commits to get the code correct and to match the current way of building plugins (i.e. version number in compiles, updated cmake, etc.), but the only bit that may be 'scary' would be the translation stuff. The translation stuff is already in use in ocpn_draw_pi and watchdog_pi and has been for quite a while now. 

Most of the changes are to the demo_pi_sample to make it work. Two normal plugins, chartdldr_pi and wmm_pi are also changed to have version numbers and use independent translation catalogs. I could resubmit this as two pull requests to break these apart.

Regards
   Jon
